bundle/edu.gemini.auxfile.workflow/src/main/java/edu/gemini/auxfile/client/AuxFileClient.java

```diff
@@ -120,6 +122,29 @@ private static void writeChunk(AuxFileChunk chunk, File destFile) throws IOExcep
         }
     }
 
+    @Override public byte[] fetchToMemory(final SPProgramID programId, final String remoteFileName) throws AuxFileException {
+
+        try (final ByteArrayOutputStream out = new ByteArrayOutputStream(FileChunkSize)) {
```

Comment:
Yes, it basically adds a `finally` which closes all the resources. Similar to `using` in C#.